src/plugins/intel_cpu/src/transformations/transformation_pipeline.cpp

```diff
@@ -427,8 +427,8 @@ void Transformations::PreLpt(const std::vector<ov::element::Type>& defaultPrecis
         manager,
         [](const_node_ptr& node) -> bool {
             const auto consumers = node->get_output_target_inputs(0);
-            return std::all_of(consumers.begin(), consumers.end(), [](const ov::Input<ov::Node>& consumer) {
-                return !ov::is_type<ov::op::v0::MatMul>(consumer.get_node());
+            return !std::all_of(consumers.begin(), consumers.end(), [](const ov::Input<ov::Node>& consumer) {
+                return ov::is_type<ov::op::v0::MatMul>(consumer.get_node());
```

Comment:
We should avoid this change since it breaks shared decompression scenario when the weights are shared not only with MatMul layers (e.g. `MatMulDecompressConvertTest3` test)

Reply:
I found that this issue has already been fixed in scope of #28401. Therefore, I reverted these changes and left only the test.